### PR TITLE
chore: retry publish web function APIs to function app & refactor create function app code

### LINKED ISSUE #567: [Bug] Add retries for azure function code upload
## Describe the bug
After deploying azure function, sometimes the function may not be available to publish code.
In this case running "func azure functionapp publish ..." will silently fail with log:
```
Publishing 'web-api' scripts to 'web-api-allyfuncappxjohj2cwcdj4c' Function App...
Can't find app with name "web-api-allyfuncappxjohj2cwcdj4c"
Successfully published 'web-api' scripts to 'web-api-allyfuncappxjohj2cwcdj4c' Function App.
```

We should check if the azure function exists & retry.

## Area

- [ ] Install script
- [ ] Update script
- [ ] Service

### PR DESCRIPTION
**Pull request by AhmedAbdoOrtiga**
#### Description of changes

(Give an overview. Add a technical description describing your changes.)

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #567
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/resource-deployment/scripts/function-app-create.sh [hunk 1]
```diff
@@ -189,14 +162,11 @@ fi
 
 installAzureFunctionsCoreTools
 
-webWorkersPackageName="web-workers"
-webApiPackageName="web-api"
-
-deployWebWorkersFunctionApp
-deployWebApiFunctionApp
+webWorkersFunctionAppName = $(deployWebAFunctionApp "web-workers-allyfuncapp" "${0%/*}/../templates/function-web-workers-app-template.json")
```
Comment (lamaks) [inline packages/resource-deployment/scripts/function-app-create.sh]: does the `var  = value` with spaces works in bash? seems this script should fail

File: packages/resource-deployment/scripts/function-app-create.sh [hunk 2]
```diff
@@ -129,45 +132,15 @@ deployWebApiArmTemplate() {
...
-    resources=$(az group deployment create \
-        --resource-group "$resourceGroupName" \
-        --template-file "$templateFilePath" \
-        --parameters namePrefix="$functionAppNamePrefix" releaseVersion="$releaseVersion" \
-        --query "properties.outputResources[].id" \
-        -o tsv)
-
-    . "${0%/*}/get-resource-name-from-resource-paths.sh" -p "Microsoft.Web/sites" -r "$resources"
-    webWorkersFunctionAppName="$resourceName"
-
```
Comment (lamaks) [inline packages/resource-deployment/scripts/function-app-create.sh]: why this got removed?

File: packages/resource-deployment/scripts/function-app-create.sh [hunk 3]
```diff
@@ -189,14 +162,11 @@ fi
...
-webApiPackageName="web-api"
-
-deployWebWorkersFunctionApp
-deployWebApiFunctionApp
+webWorkersFunctionAppName = $(deployWebAFunctionApp "web-workers-allyfuncapp" "${0%/*}/../templates/function-web-workers-app-template.json")
+publishFunctionAppScripts  "web-workers" $webWorkersFunctionAppName
 
-publishFunctionAppScripts $webApiPackageName $webApiFunctionAppName
-publishFunctionAppScripts $webWorkersPackageName $webWorkersFunctionAppName
+webApiFunctionAppName = $(deployWebAFunctionApp "web-api-allyfuncapp" "${0%/*}/../templates/function-web-api-app-template.json")
```
Comment (lamaks) [inline packages/resource-deployment/scripts/function-app-create.sh]: typo deployWebAFunctionApp?

File: packages/resource-deployment/scripts/function-app-create.sh [hunk 4]
```diff
@@ -189,14 +162,11 @@ fi
...
-webApiPackageName="web-api"
-
-deployWebWorkersFunctionApp
-deployWebApiFunctionApp
+webWorkersFunctionAppName = $(deployWebFunctionApp "web-workers-allyfuncapp" "${0%/*}/../templates/function-web-workers-app-template.json")
+publishFunctionAppScripts  "web-workers" $webWorkersFunctionAppName
 
-publishFunctionAppScripts $webApiPackageName $webApiFunctionAppName
-publishFunctionAppScripts $webWorkersPackageName $webWorkersFunctionAppName
+webApiFunctionAppName = $(deployWebFunctionApp "web-api-allyfuncapp" "${0%/*}/../templates/function-web-api-app-template.json")
```
Comment (lamaks) [inline packages/resource-deployment/scripts/function-app-create.sh]: does the var = value with spaces works in bash? seems this script should fail

File: packages/resource-deployment/scripts/function-app-create.sh (hunk 2 above)
Comment (AhmedAbdoOrtiga) [reply packages/resource-deployment/scripts/function-app-create.sh]: it's the same code, merged "deployWebWorkersArmTemplate" and "deployWebApiArmTemplate" in one function "deployWebArmTemplate"

**Pull request closed without merging**